x/interchainqueries/keeper/keeper.go

```
@@ -260,3 +263,26 @@ func (k Keeper) checkRegisteredQueryExists(ctx sdk.Context, id uint64) bool {
 
 	return store.Has(types.GetRegisteredQueryByIDKey(id))
 }
+
+func (k Keeper) CollectDeposit(ctx sdk.Context, queryInfo types.RegisteredQuery) error {
+	owner, err := queryInfo.GetOwnerAddress()
+	if err != nil {
+		panic(err.Error())
```

I don't think `panic` is good here. Why not just return an error like everywhere else?

---

But if we can not get query owner address it means that something wrong with address, not sure that this is expected situation and should be covered with exception. What do you think?